### PR TITLE
Make NodeSpec work better with Mockito

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeSpec.java [hunk 1]
```diff
@@ -454,6 +578,118 @@ public Builder parentHostname(String parentHostname) {
             return this;
         }
 
+        public Builder updateFromNodeAttributes(NodeAttributes attributes) {
+            if (attributes.getDockerImage() != null) {
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeSpec.java]: Consider
```
Optional.ofNullable(attributes.getDockerImage()).ifPresent(this::currentDockerImage);
Optional.ofNullable(attributes.getHardwareDivergence()).ifPresent(this::hardwareDivergence);
Optional.ofNullable(attributes.getRebootGeneration()).ifPresent(this::currentRebootGeneration);
Optional.ofNullable(attributes.getRestartGeneration()).ifPresent(this::currentRestartGeneration);
```

**Review by freva — APPROVED** [review]
Maybe start using the methods in `node-admin` so we can make them `private` after a PR i `hosted`?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeSpec.java (hunk 1 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeSpec.java]: Already considered

Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeSpec.java]: Fixed in PR 5696

**Pull request merged**